sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java

```diff
@@ -579,8 +606,96 @@ public void close() {
         mDatabaseManager.close();
     }
 
-    private List<Long> getLogsIds(SQLiteQueryBuilder builder, String[] selectionArgs) {
-        List<Long> result = new ArrayList<>();
+    /**
+     * Delete the oldest logs that do not fit max storage size.
+     */
+    public void deleteLogsThatNotFitMaxSize() {
+        int normalPriority = Flags.getPersistenceFlag(Flags.NORMAL, false);
+        while (getStoredDataSize() >= mDatabaseManager.getMaxSize()) {
+            if (deleteTheOldestLog(COLUMN_PRIORITY, normalPriority) == OPERATION_FAILED_FLAG) {
+                break;
+            }
+        }
+    }
+
+    private long getStoredDataSize() {
+        return mDatabaseManager.getCurrentSize() + mLargePayloadsSize;
+    }
+
+    /**
+     * Delete the log record from the database and the large payload file associated with this record if it exists.
+     *
+     * @param priorityColumn Name of priority column.
+     * @param priority Value of maximum priority of record to delete.
+     * @return Id of deleted record.
+     */
+    private long deleteTheOldestLog(@NonNull String priorityColumn, int priority) {
```

Comment:
The `priorityColumn` parameter is not needed here. Just use `DatabasePersistence.COLUMN_PRIORITY` in the method.